core/src/main/java/org/jobrunr/utils/resources/ClassPathResourceProvider.java

```diff
@@ -77,6 +80,26 @@ private Path toPath(URL url) {
             if ("wsjar".equals(uri.getScheme())) { // support for openliberty
                 uri = new URI(uri.toString().replace("wsjar", "jar"));
             }
+            if ("vfs".equals(uri.getScheme())) {    // support for Jboss/Wildfly
+                // Reflection as we cannot afford a dependency to Jboss/WildFly
+                Object virtualFile = url.openConnection().getContent();
+                Class virtualFileClass = virtualFile.getClass();
+
+                try {
+                    Method getChildrenRecursivelyMethod = virtualFileClass.getMethod("getChildrenRecursively");
+                    Method getPhysicalFileMethod = virtualFileClass.getMethod("getPhysicalFile");
+
+                    List virtualFiles = (List) getChildrenRecursivelyMethod.invoke(virtualFile);
+                    for (Object child : virtualFiles) {
+                        getPhysicalFileMethod.invoke(child);// side effect: create real-world files
+                    }
+                    File rootDir = (File) getPhysicalFileMethod.invoke(virtualFile);
+                    uri = URI.create(org.jobrunr.utils.resources.VfsFilesystemProvider.SCHEME + rootDir.toURI());
+                } catch (InvocationTargetException | NoSuchMethodException | IllegalArgumentException |
+                         IllegalAccessException e) {
```

Comment:
Can we extract everything related to VFS to a separate method?

Reply:
done